app/scripts/controllers/app-state-controller.ts

```diff
@@ -0,0 +1,867 @@
+import { LastInteractedConfirmationInfo } from '../../../shared/types/confirm';
+import { SecurityAlertResponse } from '../lib/ppom/types';
+import type {
+  Preferences,
+  PreferencesControllerGetStateAction,
+  PreferencesControllerStateChangeEvent,
+} from './preferences-controller';
+
+export type AppStateControllerState = {
+  timeoutMinutes: number;
+  connectedStatusPopoverHasBeenShown: boolean;
+  defaultHomeActiveTabName: string | null;
+  browserEnvironment: Record<string, string>;
+  popupGasPollTokens: string[];
+  notificationGasPollTokens: string[];
+  fullScreenGasPollTokens: string[];
+  recoveryPhraseReminderHasBeenShown: boolean;
+  recoveryPhraseReminderLastShown: number;
+  outdatedBrowserWarningLastShown: number | null;
+  nftsDetectionNoticeDismissed: boolean;
+  showTestnetMessageInDropdown: boolean;
+  showBetaHeader: boolean;
+  showPermissionsTour: boolean;
+  showNetworkBanner: boolean;
+  showAccountBanner: boolean;
+  trezorModel: string | null;
+  currentPopupId?: number;
+  onboardingDate: number | null;
+  lastViewedUserSurvey: number | null;
+  newPrivacyPolicyToastClickedOrClosed: boolean | null;
+  newPrivacyPolicyToastShownDate: number | null;
+  // This key is only used for checking if the user had set advancedGasFee
+  // prior to Migration 92.3 where we split out the setting to support
+  // multiple networks.
+  hadAdvancedGasFeesSetPriorToMigration92_3: boolean;
+  qrHardware: Json;
+  nftsDropdownState: Json;
+  usedNetworks: Record<string, boolean>;
+  surveyLinkLastClickedOrClosed: number | null;
+  signatureSecurityAlertResponses: Record<string, SecurityAlertResponse>;
+  // States used for displaying the changed network toast
+  switchedNetworkDetails: Record<string, string> | null;
+  switchedNetworkNeverShowMessage: boolean;
+  currentExtensionPopupId: number;
+  lastInteractedConfirmationInfo?: LastInteractedConfirmationInfo;
+  termsOfUseLastAgreed?: number;
+  snapsInstallPrivacyWarningShown?: boolean;
+  interactiveReplacementToken?: { url: string; oldRefreshToken: string };
+  noteToTraderMessage?: string;
+  custodianDeepLink?: { fromAddress: string; custodyId: string };
+};
+
+const controllerName = 'AppStateController';
+
+/**
+ * Returns the state of the {@link AppStateController}.
+ */
+export type AppStateControllerGetStateAction = {
+  type: 'AppStateController:getState';
+  handler: () => AppStateControllerState;
+};
+
+/**
+ * Actions exposed by the {@link AppStateController}.
+ */
+export type AppStateControllerActions = AppStateControllerGetStateAction;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions =
+  | AddApprovalRequest
+  | AcceptRequest
+  | PreferencesControllerGetStateAction;
+
+/**
+ * Event emitted when the state of the {@link AppStateController} changes.
+ */
+export type AppStateControllerStateChangeEvent = {
+  type: 'AppStateController:stateChange';
+  payload: [AppStateControllerState, []];
+};
```

Comment:
My understanding is that this event is not emitted through the messenger, but from EventEmitter - is that correct?